File: kotlin/snippets/Snippet-004.kt
package fr.o80.kata

class View(
    private val parentView: Screen
) {

    fun refreshIconNotification() {
        // TODO funny thing to do just for displaying a badge
        if (parentView is MainScreen || parentView is AccountsScreen) {
            iconsBlock.setNotification("", 4)
        } else {
            Accounts.getInstance().getAllAccountsList(object : Accounts.GetAccountsCalback<AccountListObject>() {
                override fun onSuccess(result: List<AccountListObject>) {
                    for (AccountsListObject accountList in result) {
                        if (!accountList.isAlreadyDiscovered()) {
                            iconsBlock.setNotification(" ", 4);
                            return;
                        }
                    }
                    iconsBlock.setNotification("", 4); // Reinit
                }

                override fun onFailure(error: Throwable) {
                    // Nothing to do
                }
            })
        }
    }

}

File: snippets/js/Snippet-004.js
import {Accounts, AccountsScreen, iconsBlock, MainScreen} from './lib/mock-library';

class View {
    constructor(parentView) {
        this.parentView = parentView;
    }

    refreshIconNotification () {
        // TODO funny thing to do just for displaying a badge
        if (this.parentView instanceof MainScreen || this.parentView instanceof AccountsScreen) {
            iconsBlock.setNotification("", 4)
        } else {
            Accounts.getInstance().getAllAccountsList(() => () => new class extends Accounts.GetAccountsCalback {
                onSuccess (result) {
                    for (const accountList of result) {
                        if (!accountList.isAlreadyDiscovered()) {
                            iconsBlock.setNotification(" ", 4);
                            return;
                        }
                    }
                    iconsBlock.setNotification("", 4); // Reinit
                }

                onFailure(error) {
                    // Nothing to do
                }
            }())
        }
    }

}